### PR TITLE
Add -ingest-storage.kafka.wait-strong-read-consistency-timeout

### PR DESCRIPTION
**Pull request by pracucci**
#### What this PR does

When the experimental ingest storage is enabled and ingesters receive a request with strong read consistency enabled, the ingester waits until the strong consistency is guaranteed. In case the ingester is lagging behind, it may take a while to reach consistency. While waiting, a querier worker slot is occupied and this may contribute the slow down the entire read path.

The way strong read consistency is currently implemented is inefficient (we shouldn't occupy a querier worker slow while waiting for it) and we'll improve it over the next few weeks. However, in the meanwhile, I don't feel comfortable having no timeout when waiting for read consistency (the only timeout we have is the query timeout, which is few minutes).

In this PR I propose to add a shorter timeout on waiting for strong read consistency in the ingester. I prefer to fail sooner rather than keeping a querier slot occupied until the query timeout expires.

#### Which issue(s) this PR fixes or relates to

N/A

#### Checklist

- [x] Tests updated.
- [ ] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Review by pstibrany — APPROVED** [review]
Makes sense, and looks like a good safety check.

**Comment by pracucci** [comment]
@tacole02 I'm merging because we're in a hurry to get this merged by EOD, but I will address any post-merge comment you may have about wording.

File: docs/sources/mimir/configure/configuration-parameters/index.md [hunk 1]
```diff
@@ -3772,6 +3772,11 @@ kafka:
   # CLI flag: -ingest-storage.kafka.producer-max-record-size-bytes
   [producer_max_record_size_bytes: <int> | default = 15983616]
 
+  # The maximum allowed for a read requests processed by an ingester to wait
+  # until strong read consistency is enforced. 0 to disable the timeout.
```
Comment (tacole02) [inline docs/sources/mimir/configure/configuration-parameters/index.md:3776]: Two small comments:

1. Line 3775. We should specify the maximum what. Time in seconds? Etc.
2. Line 3776. We should use active language here, i.e. "To disable the timeout, enter 0".

File: docs/sources/mimir/manage/mimir-runbooks/_index.md [hunk 2]
```diff
@@ -1454,13 +1454,14 @@ This alert fires when too many read-requests with strong consistency are failing
 How it **works**:
 
 - When read request asks for strong-consistency guarantee, ingester will read the last produced offset from Kafka, and wait until record with this offset is consumed.
-- If read request times out during this wait, that is considered to be a failure of request with strong-consistency.
+- If read request times out during this wait, either because of the request timeout or the configured `-ingest-storage.kafka.wait-strong-read-consistency-timeout` (whatever happens first), that is considered to be a failure of request with strong-consistency.
```
Comment (tacole02) [inline docs/sources/mimir/manage/mimir-runbooks/_index.md:1457]: We can remove "(whatever happens first)", as this is implied.

**Pull request merged**